solr/core/src/java/org/apache/solr/packagemanager/PackageManager.java

```diff
@@ -237,8 +237,10 @@ public Map<String, SolrPackageInstance> getPackagesDeployed(String collection) {
       NamedList<Object> result =
           solrClient.request(
               new GenericSolrRequest(
-                  SolrRequest.METHOD.GET,
-                  PackageUtils.getCollectionParamsPath(collection) + "/PKG_VERSIONS"));
+                      SolrRequest.METHOD.GET,
+                      PackageUtils.getCollectionParamsPath(collection) + "/PKG_VERSIONS")
+                  .setRequiresCollection(
+                      false) /* Making a collection request, but already baked into path */);
```

Comment:
interesting, is this a place we could try and improve things to not have this logic?   Seems  a shame to have a collection request and state 'but we don't require it'....?

Reply:
We could - basically that'd involve changing `PackageUtils.getCollectionParamsPath` to return a value of `/config/params` instead of the current value `/api/collections/someCollectionName/config/params`.  The method has 8 or 10 call sites, so we'd have to restructure the logic in each of those places.

It's not that big of a lift; I'm happy to go that route if it'd be your preference?  Lmk.

That said, I have a slight preference for skipping this now.  Mostly because, as you mentioned - this code really shouldn't be using GSR in any case.  No point spending time reworking the path-building logic now, if down the line we're gonna replace all this code anyways with a more appropriate SolrRequest impl.

Reply:
Do we have JIRA tickets to track removing GSR?   I think my concern is that we forget about this follow up refactoring and it never happens.   Although, maybe we *would* fix it by just having the better v2 SolrRequest impl.  Let's keep moving forward!  And not have this PR grow too much!

Reply:
We didn't, but we do now: https://issues.apache.org/jira/browse/SOLR-17136